torch/autograd/profiler.py

```diff
@@ -481,6 +481,8 @@ def _start_kineto_trace(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         if not self.enabled:
             return
+        if torch.cuda.is_available():
+            torch.cuda.synchronize()
```

Comment:
Don't have too much context but just for curiosity - does this mean there can be a perf regression when wrapping CUDA workloads under the profiler? IIUC we generally want to avoid explicit synchronizations because subsequent GPU operations will have to wait to be scheduled?

Reply:
sure, there's always a regression when using a profiler on cpu or cuda workloads, profiler should never be enabled by default on perf critical paths (only during debugging);
this synchronization happens once at the exit, to make sure that the on-device kernels of the ops that ran under profiler are in the trace